### ISSUE TITLE
Bootstrap selectpicker is hidden in a form repeater using Livewire 3

### ISSUE THREAD

**Issue by karimjammoul** [livewire/livewire#8035]
### Discussed in https://github.com/livewire/livewire/discussions/8034

<div type='discussions-op-text'>

<sup>Originally posted by **karimjammoul** February 27, 2024</sup>
This is my blade code:
                      `

        @foreach($tillAmounts as $key => $tillAmount)
        <div wire:key="tillAmount-{{ $key }}">
               <div wire:ignore>
                     <select
                          wire:model.live="tillAmounts.{{ $key }}.currency_id"
                          id="currency-{{$key}}"
                          class="selectpicker w-100"
                          required
                      >
                            @foreach($currencies as $currency)
                                <option value="{{$currency->id}}">{{$currency->name}}</option>
                            @endforeach
                      </select>
                </div>
                <div class="col-2">
                     @if($key == 0)
                          <button type="button" class="btn btn-success ms-2" wire:click="addRow">Add Amount</button>
                     @else
                          <button type="button" class="btn btn-danger ms-2"  wire:click="removeTillAmount({{ $key }})">Remove</button>
                     @endif
                 </div
          </div>
          @endforeach

`
                        
                       
and this is my component code:

`

        public function addRow()
        {
               $this->tillAmounts[] = [
                   'currency_id' => '',
               ];
               $this->dispatch('triggerLibraries');
         }

`

and here is my script:

`

            @script
            <script>

            triggerCleave()
            $('.selectpicker').selectpicker();

            Livewire.on('triggerLibraries', function () {
                $('.selectpicker').selectpicker()
                triggerCleave()
            })

            </script>
            @endscript
`

This code works fine for the selectpicker in Livewire 2, however in Livewire 3 a problem occurs when adding a new row, when I open the page for the first time the selectpicker is initialized for the first row, but when I add a second row the selectpicker in the second row is hidden, and if I added a third row the selectpicker is displayed in the second row and hidden in the third row so its like initializing all rows except the last, I don't know it could be that the dispatching events is triggered before rendering the newly added row in the blade, any help is appreciated.

Thank you</div>

**Comment by PhiloNL** [comment]
This issue has been closed as we are now using discussions to track bugs and ideas, and as there is already a discussion open we will continue investigating in that discussion.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

**Comment by adityadees** [comment]
> ### Discussed in #8034
> Originally posted by **karimjammoul** February 27, 2024 This is my blade code: `
> 
> ```
>     @foreach($tillAmounts as $key => $tillAmount)
>     <div wire:key="tillAmount-{{ $key }}">
>            <div wire:ignore>
>                  <select
>                       wire:model.live="tillAmounts.{{ $key }}.currency_id"
>                       id="currency-{{$key}}"
>                       class="selectpicker w-100"
>                       required
>                   >
>                         @foreach($currencies as $currency)
>                             <option value="{{$currency->id}}">{{$currency->name}}</option>
>                         @endforeach
>                   </select>
>             </div>
>             <div>
>                  @if($key == 0)
>                       <button type="button" class="btn btn-success ms-2" wire:click="addRow">Add Amount</button>
>                  @endif
>              </div
>       </div>
>       @endforeach
> ```
> 
> `
> 
> and this is my component code:
> 
> `
> 
> ```
>     public function addRow()
>     {
>            $this->tillAmounts[] = [
>                'currency_id' => '',
>            ];
>            $this->dispatch('triggerLibraries');
>      }
> ```
> 
> `
> 
> and here is my script:
> 
> `
> 
> ```
>         @script
>         <script>
> 
> 
>         $('.selectpicker').selectpicker();
> 
>         Livewire.on('triggerLibraries', function () {
>             $('.selectpicker').selectpicker()
>         })
> 
>         </script>
>         @endscript
> ```
> 
> `
> 
> This code works fine for the selectpicker in Livewire 2, however in Livewire 3 a problem occurs when adding a new row, when I open the page for the first time the selectpicker is initialized for the first row, but when I add a second row the selectpicker in the second row is hidden, and if I added a third row the selectpicker is displayed in the second row and hidden in the third row so its like initializing all rows except the last, I don't know it could be that the dispatched "triggerLibraries" event is triggered before rendering the newly added row in the blade, any help is appreciated.
> 
> Thank you

yes the same, no idea here. but from livewire 2 I think more better than livewire 3.
not sure I think documentation not explained well or some breaking changes make this not working anymore.

**Comment by karimjammoul** [comment]
> > ### Discussed in #8034
> > Originally posted by **karimjammoul** February 27, 2024 This is my blade code: `
> > ```
> >     @foreach($tillAmounts as $key => $tillAmount)
> >     <div wire:key="tillAmount-{{ $key }}">
> >            <div wire:ignore>
> >                  <select
> >                       wire:model.live="tillAmounts.{{ $key }}.currency_id"
> >                       id="currency-{{$key}}"
> >                       class="selectpicker w-100"
> >                       required
> >                   >
> >                         @foreach($currencies as $currency)
> >                             <option value="{{$currency->id}}">{{$currency->name}}</option>
> >                         @endforeach
> >                   </select>
> >             </div>
> >             <div>
> >                  @if($key == 0)
> >                       <button type="button" class="btn btn-success ms-2" wire:click="addRow">Add Amount</button>
> >                  @endif
> >              </div
> >       </div>
> >       @endforeach
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > `
> > and this is my component code:
> > `
> > ```
> >     public function addRow()
> >     {
> >            $this->tillAmounts[] = [
> >                'currency_id' => '',
> >            ];
> >            $this->dispatch('triggerLibraries');
> >      }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > `
> > and here is my script:
> > `
> > ```
> >         @script
> >         <script>
> > 
> > 
> >         $('.selectpicker').selectpicker();
> > 
> >         Livewire.on('triggerLibraries', function () {
> >             $('.selectpicker').selectpicker()
> >         })
> > 
> >         </script>
> >         @endscript
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > `
> > This code works fine for the selectpicker in Livewire 2, however in Livewire 3 a problem occurs when adding a new row, when I open the page for the first time the selectpicker is initialized for the first row, but when I add a second row the selectpicker in the second row is hidden, and if I added a third row the selectpicker is displayed in the second row and hidden in the third row so its like initializing all rows except the last, I don't know it could be that the dispatched "triggerLibraries" event is triggered before rendering the newly added row in the blade, any help is appreciated.
> > Thank you
> 
> yes the same, no idea here. but from livewire 2 I think more better than livewire 3. not sure I think documentation not explained well or some breaking changes make this not working anymore.

I fixed it using this:

`

@script
<script>
        // to trigger the selectpicker when opening the page
        $('.selectpicker').selectpicker();

        // to trigger the selectpicker when adding a new row
        Livewire.hook('morph.added', ({ el }) => {
            $('.selectpicker').selectpicker();
        })
</script>
@endscript
`
